src/stan/callbacks/json_writer.hpp

```diff
@@ -46,6 +46,7 @@ class json_writer {
   void write_record_comma_if_needed() {
     if (record_depth_ > 0 && record_needs_comma_) {
       *output_ << ",";
+      record_needs_comma_ = false;
```

Comment:
This can be 

```cpp
record_needs_comma_ = record_needs_comma_ && record_depth_ > 0;
if (record_needs_comma_) {
  *output_ << ",";
}
```

But then I wonder why is the variable is being set?  (I also reordered the conjunction---there is no `&&=`, alas)

Reply:
it gets set on line 293.